packages/@aws-cdk/core/lib/asset-staging.ts

```diff
@@ -102,14 +113,27 @@ export class AssetStaging extends Construct {
       const bundlingStacks: string[] = this.node.tryGetContext(cxapi.BUNDLING_STACKS) ?? ['*'];
       const runBundling = bundlingStacks.includes(Stack.of(this).stackName) || bundlingStacks.includes('*');
       if (runBundling) {
-        // Determine the source hash in advance of bundling if the asset hash type
-        // is SOURCE so that the bundler can opt to re-use its previous output.
-        const sourceHash = hashType === AssetHashType.SOURCE
-          ? this.calculateHash(hashType, props.assetHash, props.bundling)
-          : undefined;
-
-        this.bundleDir = this.bundle(props.bundling, outdir, sourceHash);
-        this.assetHash = sourceHash ?? this.calculateHash(hashType, props.assetHash, props.bundling);
+        // Check if we already bundled this source path in this App (e.g. the same
+        // asset is used in multiple stacks). In this case we can completely skip
+        // file system and bundling operations.
+        this.cacheHash = crypto.createHash('sha256')
```

Comment:
Maybe call this `cacheKey`, otherwise I get confused with all the occurrences of the word `hash` here.

Why does it need to be a member?

Reply:
Also feels like it would be cleaner if this calculation was in a function somewhere.